pkgs/hooks_runner/test_data/download_assets/hook/build.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:io';
+
+import 'package:code_assets/code_assets.dart';
+import 'package:hooks/hooks.dart';
+
+void main(List<String> args) async {
+  await build(args, (input, output) async {
+    final file = File.fromUri(input.outputDirectory.resolve('testfile'));
+
+    final client = HttpClient()
+      ..findProxy = HttpClient.findProxyFromEnvironment;
+    // This request should fail outside of proxies.
+    final request = await client.get(
+      'testing_proxy.dart.dev',
+      80,
+      'test_asset',
+    );
+    final response = await request.close();
+
+    if (response.statusCode != 200) {
+      throw Exception('Could not fetch asset');
+    }
+
+    await response.pipe(file.openWrite());
+
+    output.assets.code.add(
```

Comment:
We can probably leave the asset out. Failing the hook if there was an error should be enough. The integration test can simply check that the hook didn't fail.